libs/wingsdk/src/cloud/secret.md

```diff
@@ -0,0 +1,79 @@
+---
+title: cloud.Secret 
+id: secret
+description: A built-in resource for securely storing secrets in the cloud.
+keywords: [Wing reference, Wing language, language, Wing sdk, Wing programming language, secrets]
+---
+
+The `cloud.Secret` class represents a secret value (like an API key, certificate, etc.) that is securely stored in the cloud.
+
+Secrets are encrypted at rest and in transit, and are only decrypted when they are used in a task.
+Storing a secret allows you to use the value in different compute tasks while only having to rotate or revoke it in one place.
+
+## Usage
+
+### Creating a secret
+
+```js
+bring cloud;
+
+let secret = new cloud.Secret(
+  name: "my-secret", // optional, defaults to a generated name
+);
+```
+
+Before deploying your application, you will be expected to store the secret value in a secure place according to the target-specific instructions below.
+
+### Using a secret
+
+```js
+bring cloud;
+
+let secret = new cloud.Secret(
+  name: "my-api-key",
+);
+
+new cloud.Function(inflight () => {
+  let secretValue = secret.value(); // retrieve the secret as a `str` value
+  let secretValueAsJson = secret.valueJson(); // retrieve the secret as a `Json` value
+});
+```
+
+## Target-specific details
+
+### Simulator (`sim`)
+
+When using a secret in Wing's simulator, a secrets file must be added to your home directory at `~/.wing/secrets.json`.
+Secrets should be saved in a JSON format:
+
+```json
+// secrets.json
+{
+  "my-api-key": "1234567890"
+}
+```
+
+### AWS (`tf-aws` and `awscdk`)
+
+AWS implementations of `cloud.Secret` use AWS Secrets Manager.
+Before deploying your application, you must create a secret in the AWS account with the same name as the secret in your Wing application.
+You can do this using the AWS CLI:
+
+```bash
+aws secretsmanager create-secret --name my-api-key --secret-string 1234567890
```

Comment:
Can i set it from my code? 
for example, set a random string as a secret

Reply:
@schosterbarak It's not available right now but it sounds like something that could be added. 🙂 AWS CDK supports creating a string from plaintext or generating it randomly so its definitely feasible: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_secretsmanager-readme.html#create-a-new-secret-in-a-stack

Of course, we want to design the APIs in a way that encourages best practices, since you want to avoid ending up with plaintext secrets in your source code OR in the infrastructure config files created by Wing. This guide describes some of the pros and cons of different approaches in Terraform for example: https://blog.gruntwork.io/a-comprehensive-guide-to-managing-secrets-in-your-terraform-code-1d586955ace1